src/instructlab/configuration.py

```diff
@@ -29,25 +31,63 @@
 # Local
 from . import log
 
+
+class STORAGE_DIR_NAMES:
+    ILAB = "instructlab"
+    DATASETS = "datasets"
+    CHECKPOINTS = "checkpoints"
+    MODELS = "models"
+    TAXONOMY = "taxonomy"
+    INTERNAL = (
+        "internal"  # for storing all ilab-internal files the user doesn't need to see
+    )
+
+
+def get_home_dir(*subdir: str) -> str:
+    return os.path.join(str(Path.home()), *subdir)
+
+
+def get_home_env_path(env: str, suffix: str, *home_paths: str) -> str:
+    """Neatly formats the environment paths for the user home & storage directories"""
+    home_path = os.getenv(env)
+    full_path = home_path if home_path else get_home_dir(*home_paths)
+    return os.path.join(full_path, suffix)
+
+
+ILAB_CONFIG_HOME = get_home_env_path(
+    "XDG_CONFIG_HOME", STORAGE_DIR_NAMES.ILAB, ".config"
+)
+ILAB_DATA_HOME = get_home_env_path(
+    "XDG_DATA_HOME", STORAGE_DIR_NAMES.ILAB, ".local", "share"
+)
+
 DEFAULT_API_KEY = "no_api_key"
-DEFAULT_CONFIG = "config.yaml"
+DEFAULT_CONFIG = os.path.join(ILAB_CONFIG_HOME, "config.yaml")
 # TODO: Consolidate --model and --model-path into one --model-path flag since we always need a path now
 DEFAULT_MODEL_OLD = "merlinite-7b-lab-Q4_K_M"
 DEFAULT_MODEL = "models/merlinite-7b-lab-Q4_K_M.gguf"
-DEFAULT_MODEL_PATH = "models/merlinite-7b-lab-Q4_K_M.gguf"
+DEFAULT_MODEL_DIR = os.path.join(ILAB_DATA_HOME, STORAGE_DIR_NAMES.MODELS)
+DEFAULT_MODEL_PATH = os.path.join(DEFAULT_MODEL_DIR, "merlinite-7b-lab-Q4_K_M.gguf")
 DEFAULT_MODEL_REPO = "instructlab/granite-7b-lab"
 DEFAULT_JUDGE_MODEL_MT = "prometheus-eval/prometheus-8x7b-v2.0"
 DEFAULT_EVAL_PATH = "eval_data"
 DEFAULT_TAXONOMY_REPO = "https://github.com/instructlab/taxonomy.git"
-DEFAULT_TAXONOMY_PATH = "taxonomy"
+DEFAULT_TAXONOMY_PATH = os.path.join(ILAB_DATA_HOME, STORAGE_DIR_NAMES.TAXONOMY)
 DEFAULT_TAXONOMY_BASE = "origin/main"
+DEFAULT_CHECKPOINTS_DIR = os.path.join(ILAB_DATA_HOME, STORAGE_DIR_NAMES.CHECKPOINTS)
+DEFAULT_DATASET_DIR = os.path.join(ILAB_DATA_HOME, STORAGE_DIR_NAMES.DATASETS)
+DEFAULT_INTERNAL_DIR = os.path.join(ILAB_DATA_HOME, STORAGE_DIR_NAMES.INTERNAL)
```

Comment:
Are these really user data files? At least the checkpoints directory probably belongs into user cache dir.